src/main.rs

```diff
@@ -154,10 +202,20 @@ fn run() -> Result<ExitCode> {
         None
     };
 
-    let command = if let Some(args) = matches.values_of("exec") {
-        Some(CommandTemplate::new(args))
-    } else if let Some(args) = matches.values_of("exec-batch") {
-        Some(CommandTemplate::new_batch(args)?)
+    let mut commands: Vec<CommandTemplate> = vec![];
+
+    if matches.is_present("exec") {
+        let ranges = ContinuousRangeIterator::new(matches.indices_of("exec").unwrap().peekable());
+        let args: Vec<&str> = matches.values_of("exec").unwrap().collect();
+        for (start, end) in ranges {
+            commands.push(CommandTemplate::new(&args[start..end]));
+        }
+    } else if matches.is_present("exec-batch") {
+        let ranges = ContinuousRangeIterator::new(matches.indices_of("exec-batch").unwrap().peekable());
+        let args: Vec<&str> = matches.values_of("exec-batch").unwrap().collect();
+        for (start, end) in ranges {
+            commands.push(CommandTemplate::new_batch(&args[start..end])?);
```

Comment:
Its a shame to duplicate this code for both exec and exec batch. This is a result of my borrow checker woes. Any ideas?

Reply:
I don't see any problem in moving this code to a separate function or lambda except for the `::new(…)` and `::new_batch` calls. One way to solve it would be to merge these two functions into a single one (`::new`) that takes a `ExecutionMode` argument.